local build script to build the docker image

ydubreuil opened the pull request:
Simulate the Jenkins Pipeline docker step locally in a shell script so BlueOcean is built like in Jenkins

@reviewbybees

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

michaelneale commented on Jenkinsfile:
```diff
@@ -1,7 +1,7 @@
   deleteDir()
   checkout scm
-  
+
```
whitespace changes for the Jenkinsfile are a little annoying

michaelneale commented:
Can you update this to master (next week) and try again?

ydubreuil commented:
@michaelneale rebased on master and removed the irrelevant `Jenkinsfile` modifications

michaelneale commented:
I like it. Is build.sh the right name? If so, 🐝  from me - otherwise a better name would be also good.

michaelneale commented:
@ydubreuil hrm... I think this should tie into also building a complete docker image - which is https://github.com/jenkinsci/blueocean-plugin/pull/244 - being fixed. Perhaps this script should also have an option to do that? WDYT?

ydubreuil commented:
@michaelneale yes, building an image in the end makes total sense, actually, I did it in the first place before removing it while pushing. I didn't want to build an image every time. I'll add an option to build the image on demand :)

tfennelly commented:
Cool stuff @ydubreuil !!! So this is building a docker image? Maybe the name of the script should reflect that? Also, a few comments would help too I think e.g. if there are command line params ... examples of how to run it etc etc

michaelneale commented:
yeah this is most useful - increasingly I like the idea of being able to reproduce things with simple scripts as much as possible. For some value of "simple script" ;)

seems some minor conflicts now

ydubreuil commented:
Addressed most comments I hope, there's a online help now which explains what this script is for. And rebased against master

I didn't try it on MacOS, and I'm afraid it can fails because it requires to share workspace folder with the build container (same requirement as the Pipeline Docker plugin btw).

ydubreuil commented:
Um, regarding the script name, it can probably be renamed `build-in-docker.sh`

michaelneale commented:
@ydubreuil it would likely have better success with the new "docker for OSX beta" which does proper bind mounting (unlike the virtual box based ones) - docker pipeline certainly works well on osx now with that. 

Yes, go ahead and rename it, why not!

ydubreuil commented:
@michaelneale script renamed, could you merge?

michaelneale commented:
🐝

michaelneale commented:
thanks @ydubreuil. Done.
